### PR TITLE
fix(logs): Added and refactored audience and feature variable evaluation logs

### PR DESCRIPTION
**Pull request by mnoman09**
## Summary
- Added and updated missing audience evaluation logs.

### Test plan
added unit tests must pass and all FSC scenarios should pass

### CODE REVIEW

File: OptimizelySDK.Tests/UtilsTests/ExperimentUtilsTest.cs [hunk 1]
```diff
@@ -126,7 +126,7 @@ public void TestIsUserInExperimentReturnsTrueIfAllAudiencesInANDConditionPass()
                 { "favorite_ice_cream", "walls" }
             };
 
-            Assert.True(ExperimentUtils.IsUserInExperiment(Config, experiment, userAttributes, Logger));
+            Assert.True(ExperimentUtils.IsUserInExperiment(Config, experiment, userAttributes, "experiment", experiment.Key, Logger));
```
Comment (msohailhussain) [inline OptimizelySDK.Tests/UtilsTests/ExperimentUtilsTest.cs:129]: `IsUserInExperiment` Please rename it.

File: OptimizelySDK/Bucketing/DecisionService.cs [hunk 2]
```diff
@@ -120,7 +120,7 @@ public virtual Variation GetVariation(Experiment experiment, string userId, Proj
                 }
             }
 
-            if (ExperimentUtils.IsUserInExperiment(config, experiment, filteredAttributes, Logger))
+            if (ExperimentUtils.IsUserInExperiment(config, experiment, filteredAttributes, "experiment", experiment.Key, Logger))
```
Comment (msohailhussain) [inline OptimizelySDK/Bucketing/DecisionService.cs:123]: rename it.

Comment (msohailhussain) [reply OptimizelySDK/Bucketing/DecisionService.cs:123]: Please use const `experiment`, don't pass string as it is.

File: OptimizelySDK/Optimizely.cs [hunk 3]
```diff
@@ -489,9 +489,9 @@ public virtual bool IsFeatureEnabled(string featureKey, string userId, UserAttri
             }
 
             if (featureEnabled == true)
-                Logger.Log(LogLevel.INFO, $@"Feature flag ""{featureKey}"" is enabled for user ""{userId}"".");
+                Logger.Log(LogLevel.INFO, $@"Feature ""{featureKey}"" is enabled for user ""{userId}"".");
```
Comment (msohailhussain) [inline OptimizelySDK/Optimizely.cs:492]: Following logs doesn't seem correct to me.

File: OptimizelySDK/Optimizely.cs [hunk 4]
```diff
@@ -569,11 +569,11 @@ public virtual bool IsFeatureEnabled(string featureKey, string userId, UserAttri
                     if (variation.FeatureEnabled == true)
                     {
                         variableValue = featureVariableUsageInstance.Value;
-                        Logger.Log(LogLevel.INFO, $@"Returning variable value ""{variableValue}"" for variation ""{variation.Key}"" of feature flag ""{featureKey}"".");
+                        Logger.Log(LogLevel.INFO, $@"Feature ""{featureKey}"" is enabled for user ""{userId}"".");
                     }
                     else
                     {
-                        Logger.Log(LogLevel.INFO, $@"Feature ""{featureKey}"" is not enabled for user {userId}. Returning default value for variable ""{variableKey}"".");
+                        Logger.Log(LogLevel.INFO, $@"Feature ""{featureKey}"" is not enabled for user {userId}. Returning the default variable value ""{variableValue}"".");
```
Comment (msohailhussain) [inline OptimizelySDK/Optimizely.cs:576]:
```self.logger.info(
                    'Feature "%s" is not enabled for user "%s". '
                    'Returning the default variable value "%s".' % (feature_key, user_id, variable_value)
                )```

File: OptimizelySDK/Optimizely.cs [hunk 5]
```diff
@@ -569,11 +569,11 @@ public virtual bool IsFeatureEnabled(string featureKey, string userId, UserAttri
                     if (variation.FeatureEnabled == true)
                     {
                         variableValue = featureVariableUsageInstance.Value;
-                        Logger.Log(LogLevel.INFO, $@"Returning variable value ""{variableValue}"" for variation ""{variation.Key}"" of feature flag ""{featureKey}"".");
+                        Logger.Log(LogLevel.INFO, $@"Feature ""{featureKey}"" is enabled for user ""{userId}"".");
```
Comment (msohailhussain) [inline OptimizelySDK/Optimizely.cs:572]: self.logger.info(
                    'Got variable value "%s" for variable "%s" of feature flag "%s".'
                    % (variable_value, variable_key, feature_key)
                )

File: OptimizelySDK.Tests/UtilsTests/ValidatorTest.cs [hunk 6]
```diff
@@ -81,15 +81,15 @@ public void TestValidateJsonSchemaNoJsonContent()
         [Test]
         public void TestIsUserInExperimentNoAudienceUsedInExperiment()
```
Comment (aliabbasrizvi) [inline OptimizelySDK.Tests/UtilsTests/ValidatorTest.cs:82]: The name of the method has changed to: `DoesUserMeetAudienceConditions`. Please update test names accordingly.

File: OptimizelySDK/Bucketing/DecisionService.cs [hunk 7]
```diff
@@ -35,6 +35,12 @@ namespace OptimizelySDK.Bucketing
     /// </summary>
     public class DecisionService
     {
+        /// <summary>
+        /// String constant representing audience for log.
+        /// </summary>
+        public const string AUDIENCE_FOR_EXPERIMENT = "experiment";
+        public const string AUDIENCE_FOR_RULE = "rule";
```
Comment (aliabbasrizvi) [inline OptimizelySDK/Bucketing/DecisionService.cs:42]: These variable names and the docstring above it do not make sense.

File: OptimizelySDK/Utils/ExperimentUtils.cs [hunk 8]
```diff
@@ -43,7 +43,12 @@ public static bool IsExperimentActive(Experiment experiment, ILogger logger)
         /// <param name="experiment">Experiment Entity representing the experiment</param>
         /// <param name="userAttributes">Attributes of the user. Defaults to empty attributes array if not provided</param>
         /// <returns>true if the user meets audience conditions to be in experiment, false otherwise.</returns>
-        public static bool IsUserInExperiment(ProjectConfig config, Experiment experiment, UserAttributes userAttributes, ILogger logger)
+        public static bool DoesUserMeetAudienceConditions(ProjectConfig config,
+            Experiment experiment,
```
Comment (aliabbasrizvi) [inline OptimizelySDK/Utils/ExperimentUtils.cs:47]: Parameter descriptions above need to be updated.

File: OptimizelySDK/Utils/ExperimentUtils.cs [hunk 9]
```diff
@@ -43,7 +43,12 @@ public static bool IsExperimentActive(Experiment experiment, ILogger logger)
...
+            UserAttributes userAttributes,
+            string audienceFor,
```
Comment (aliabbasrizvi) [inline OptimizelySDK/Utils/ExperimentUtils.cs:49]: `audienceFor`? Why not something like `loggingKeyType`?

File: OptimizelySDK/Utils/ExperimentUtils.cs [hunk 10]
```diff
@@ -42,11 +42,13 @@ public static bool IsExperimentActive(Experiment experiment, ILogger logger)
         /// <param name="config">ProjectConfig Configuration for the project</param>
         /// <param name="experiment">Experiment Entity representing the experiment</param>
         /// <param name="userAttributes">Attributes of the user. Defaults to empty attributes array if not provided</param>
+        /// <param name="loggingKeyType">It can be either experiment or rule.</param>
+        /// <param name="loggingKey">In case of loggingKeyType is experiment it will be experiment key or else it will be rule number.</param>
```
Comment (aliabbasrizvi) [inline OptimizelySDK/Utils/ExperimentUtils.cs:46]: nit. `In case loggingKeyType is experiment`

**Pull request merged**